### PR TITLE
AADS, on automl, when I launch python train.py, I get the model evaluation with standard metrics

### PR DESCRIPTION
**Pull request by FlorianLeRoyKili**
Image Segmentation TBD later

### CODE REVIEW

File: kiliautoml/models/_base_model.py
Comment (PierreLeveau) [inline kiliautoml/models/_base_model.py:43]: You should keep `DictTrainingInfos` and define it as `DictTrainingInfos= Dict[str, Any]`

File: kiliautoml/models/_hugging_face_named_entity_recognition_model.py
Comment (PierreLeveau) [inline kiliautoml/models/_hugging_face_named_entity_recognition_model.py:197]: put the evaluation part in another private method

**Review by PierreLeveau — CHANGES_REQUESTED** [review]

File: commands/train.py
Comment (FlorianLeRoyKili) [reply commands/train.py:232]: I put : 
`for evaluation in model_evaluations:
        model_evaluation = pd.DataFrame(evaluation[1]).transpose()
        print(tabulate(model_evaluation, headers=[evaluation[0]] + list(model_evaluation.columns)))`
What do you think ?

File: kiliautoml/models/_ultralytics_object_detection.py
Comment (FlorianLeRoyKili) [reply kiliautoml/models/_ultralytics_object_detection.py:239]: Yes it was not used at all.

File: tests/e2e/fixtures/img_class_project_fixture.json
Comment (FlorianLeRoyKili) [reply tests/e2e/fixtures/img_class_project_fixture.json:1]: Yes, the other project was on staging.

File: commands/train.py
Comment (FlorianLeRoyKili) [reply commands/train.py]: I manage to remove pandas. Actually, even if it was not in the requirements, it was already used in object detection to read the results.csv
I changed it to use csv (even if pandas is a dependency of yolo so if you do object detection, you have to import it anyhow)

File: kiliautoml/models/_hugging_face_named_entity_recognition_model.py
Comment (PierreLeveau) [inline kiliautoml/models/_hugging_face_named_entity_recognition_model.py:496]: evaluation

File: kiliautoml/models/_hugging_face_text_classification_model.py
Comment (PierreLeveau) [inline kiliautoml/models/_hugging_face_text_classification_model.py:135]: `dict(sorted(model_evaluation.items()))`

**Review by PierreLeveau — CHANGES_REQUESTED** [review]

File: kiliautoml/models/_hugging_face_named_entity_recognition_model.py
Comment (FlorianLeRoyKili) [reply kiliautoml/models/_hugging_face_named_entity_recognition_model.py:98]: Unfortunately, I don't think that's possible without recoding an entire HuggingFace Trainer (as compute_metrics is used inside trainer.evaluate where I cannot add parameters...).
I can try if you think it is worth it but that will take a couple of hours.

File: kiliautoml/models/_hugging_face_named_entity_recognition_model.py
Comment (PierreLeveau) [inline kiliautoml/models/_hugging_face_named_entity_recognition_model.py:199]: forgot this one: `dict(sorted(model_evaluation.items()))`

File: kiliautoml/utils/pytorchvision/trainer.py
Comment (PierreLeveau) [inline kiliautoml/utils/pytorchvision/trainer.py:45]: won't work, this will make `best_val_precision, best_val_recall,  best_val_f1` have the same values since they are pointing towards the same list. Same below

**Review by PierreLeveau — CHANGES_REQUESTED** [review]

File: kiliautoml/models/_pytorchvision_image_classification.py
Comment (PierreLeveau) [inline kiliautoml/models/_pytorchvision_image_classification.py:107]: note for later: this should be generalized to all the tasks, with a common method (for splitting and checking the label repartition)

Comment (FlorianLeRoyKili) [reply kiliautoml/models/_pytorchvision_image_classification.py:107]: yes. I think you are right. Next week's workshop with Raphaël will help us decide what to choose.

File: kiliautoml/utils/pytorchvision/trainer.py
Comment (PierreLeveau) [inline kiliautoml/utils/pytorchvision/trainer.py:44]: Sorry I missed that: that method is now way to big. I suggest a Metric object that would be initialized, updated, then rendered when the method returns.

**Review by PierreLeveau — APPROVED** [review]

File: kiliautoml/utils/pytorchvision/trainer.py
Comment (FlorianLeRoyKili) [reply kiliautoml/utils/pytorchvision/trainer.py:44]: I can put one dictionary for train metrics and one for val metrics?

**Comment by PierreLeveau** [comment]
I'll let @crsegerie have another review before merging

File: kiliautoml/models/_hugging_face_named_entity_recognition_model.py
Comment (crsegerie) [inline kiliautoml/models/_hugging_face_named_entity_recognition_model.py:114]: thank you

File: kiliautoml/models/_hugging_face_named_entity_recognition_model.py
Comment (crsegerie) [inline kiliautoml/models/_hugging_face_named_entity_recognition_model.py:176]: -100 ?

File: kiliautoml/models/_hugging_face_named_entity_recognition_model.py
Comment (crsegerie) [inline kiliautoml/models/_hugging_face_named_entity_recognition_model.py:298]: We should create an Asset class, and this kind of logic should be managed by a method of the class. But ok for now

File: kiliautoml/models/_hugging_face_named_entity_recognition_model.py
Comment (crsegerie) [inline kiliautoml/models/_hugging_face_named_entity_recognition_model.py:497]: Why there is no argument here?

File: kiliautoml/models/_hugging_face_named_entity_recognition_model.py
Comment (crsegerie) [inline kiliautoml/models/_hugging_face_named_entity_recognition_model.py:501]: Those kinds of function should be a method in the class Job

File: kiliautoml/models/_hugging_face_text_classification_model.py
Comment (crsegerie) [inline kiliautoml/models/_hugging_face_text_classification_model.py:302]: I don't understand why sometimes there is a [-1] and sometimes not

File: kiliautoml/models/_ultralytics_object_detection.py
Comment (crsegerie) [inline kiliautoml/models/_ultralytics_object_detection.py:234]: good catch

File: kiliautoml/utils/helpers.py
Comment (crsegerie) [inline kiliautoml/utils/helpers.py:314]: separate this tuple into 2 arguments

File: kiliautoml/utils/helpers.py
Comment (crsegerie) [inline kiliautoml/utils/helpers.py:335]: why not using pandas ?

File: kiliautoml/utils/pytorchvision/trainer.py
Comment (crsegerie) [inline kiliautoml/utils/pytorchvision/trainer.py:155]: why np.append?

File: kiliautoml/utils/pytorchvision/trainer.py
Comment (crsegerie) [inline kiliautoml/utils/pytorchvision/trainer.py:160]: zero_division=np.nan  is better no ?

File: tests/cmd/generate_image_classification_project_mock_data.py
Comment (crsegerie) [inline tests/cmd/generate_image_classification_project_mock_data.py:1]: I have also created a system in my mr to generate mock data...

File: kiliautoml/models/_hugging_face_named_entity_recognition_model.py
Comment (FlorianLeRoyKili) [reply kiliautoml/models/_hugging_face_named_entity_recognition_model.py:176]: add comments

File: kiliautoml/models/_hugging_face_named_entity_recognition_model.py
Comment (FlorianLeRoyKili) [reply kiliautoml/models/_hugging_face_named_entity_recognition_model.py:497]: add trainer.eval_dataset

File: kiliautoml/models/_hugging_face_named_entity_recognition_model.py
Comment (FlorianLeRoyKili) [reply kiliautoml/models/_hugging_face_named_entity_recognition_model.py:501]: ticket to create a Class Job

File: kiliautoml/models/_hugging_face_text_classification_model.py
Comment (FlorianLeRoyKili) [reply kiliautoml/models/_hugging_face_text_classification_model.py:302]: -1 correspond to the average metric (add comment)

Comment (FlorianLeRoyKili) [reply kiliautoml/models/_hugging_face_text_classification_model.py:302]: add Dataclass

**Review by crsegerie — APPROVED** [review]

**Pull request merged**
